cleaning up styles, adding functionality for slide controls and easin…

pcsandford opened the pull request:
…g library

@stulees @mmeiers please review. I have tidied up the style & index file and added in the functionality for the slides as well as a jshintrc and jshintignore to ignore the jquery easing library file I included for the bounce effect on the slides.

Stuart, I noticed that when the google sheet component refreshes the new slide data is added on top of the old ones. I compared my code to Donna's for the donor-wall and I can't see where she is removing the old slides so I keep getting errors when I try to add in a function to do so. Ideally it would happen at the end of a cycle when the counter is reset to 0 so I will keep trying, but if you see anything obvious I am missing that would be awesome. 

Thanks!

pcsandford committed:
cleaning up styles, adding functionality for slide controls and easing library,adding svg-inject through bower, adding animation to svg and adding README

stulees commented:
@pcsandford The handler for "rise-google-sheet-response" always calls `displayEvents();` in it, so perhaps you need to always clear particular slide content prior to calling that `displayEvents()` to avoid content being added over top? I'm not familiar enough with this theme to know for sure. Something I also noticed is, why is code from lines 96 - 105 wrapped inside `(document).ready()` while also residing in the `displayEvents` function? It seems out of place. Thanks

stulees commented:
@pcsandford Also, where did the jquery easing file come from? We should avoid adding 3rd party plugin code directly in the repo. Use Bower if possible, or CDN. Failing that, we should be adding it to our bucket on Amazon. Could Greensock be used instead?

pcsandford commented:
@stulees I added it in just to add 'bounce' effect to the end of the movement of the slide, but the more I watch it the less I like it so I think I will take it out for now and look to greensock in future for sure, thanks! I had wrapped it in there to use jQuery but I meant to ask if that was correct (or needed?). I had it in the function to avoid adding the applicable variables to the global scope as donna's structure had everything in private functions. 

In terms of clearing the slides I had assumed I would find a function doing this in the donor-wall script (as that where this is taken from, and was how I had initially set it up) but I couldn't see it, so I will set up a function to do this before displayEvents() is called. Thanks!

stulees commented:
@pcsandford Is the `$(document).ready()` necessary though? The source code is not imported within the `<head>` it's at the bottom of `<body>`, so it's most likely that the `$(document).ready()` is not required. 

Regarding clearing the slides, was a visual refresh working correctly in the donor-wall script that you took this from? If so, I'd recommend stepping through that code and understanding exactly how the content gets applied and updated based on `rise-google-sheet` firing "rise-google-sheet-response", which it will do initially and on every refresh interval of whatever the refresh value is set to. Thanks

pcsandford commented:
@stulees thanks! I will do that, and I will sort out the $(document).ready() in the next commit too. Thanks!

pcsandford commented:
@stulees please review. I have removed the easing js file and now using a cdn as you suggested. I have modified the js to remove the document.ready function and fix the movement issues. Also added more to the keyframes for the animation and modified the package.json file to update the gulp-min as you fixed for the event theme. Will squash commits when ready. Thanks!

stulees commented on src/js/spreadsheet-events.js:
```diff
@@ -83,21 +94,39 @@ RiseVision.SpreadsheetEvents = (function() {
       addEvents(e.detail.cells);
       displayEvents();
+      counter = 0;
+      if (slideMovement === true) {
```
@pcsandford Curious, if you're clearing the interval here, does it ever need to be started again?

stulees commented:
@pcsandford One comment in the code, otherwise :+1: Are you going to be implementing the SVG injector in a future PR? I think it's really important we don't have SVG code in the HTML. Thanks

pcsandford commented:
@stulees good point re interval, I will check it out. Also, definitely will use an SVG injector - may check with Mat if there is one he prefers so we can set a standard so will try to get to that today. I just wanted to get the gulp build task running so that there is something live to review but IU sort of took over. Thanks!

mmeiers commented:
@pcsandford Sorry I know im late to this. Might not matter if you are removing the bounce effect, but should we standardize on animate.css for these types of effects? That way it would match the way playlist component does transitions (with css)

pcsandford commented:
@stulees @mmeiers please review. I have added in SVG via bower and added in the necessary script file. jshint didn't like it as the function started with a capital so it wanted to define it as a new constructor, but as this code was taken from the repo I thought that it would be confusing to alter it and so added svg.js to the .jshintignore file before running build. 

I made the changes to the JS file to remove the code that cleared the interval as it is now only being called once. 

I have added in the repo along with images. 

@mmeiers I think that standardizing is a good idea, but when I tried to use the cdn for animate.css it gave me an error. I couldn't see from the repo how it was brought in for the 'fade-in' effects in the playlist component but if you think that the current effect is too far off what bounceInLeft would produce I can look into this more.

Thanks!

mmeiers commented:
@pcsandford ya dont worry about it for now. Now that I think about I dont think it was part of the playlist component. But the playlist component lets you leverage css keyframes. So I probably just downloaded the animate.css and linked it when i did it last.

pcsandford commented:
@mmeiers ah cool, that makes sense. Thanks!

stulees commented:
@pcsandford I think you're missing the inclusion of the SVGInjector in your bower file as I see the import for the source code `"components/svg-injector/svg-injector.js"` but no change to bower. 

Do you need a separate `svg.js` file? That code looks like it could go into main.js or spreadsheet-events.js `init()` function. 

I don't think you should have jshint ignore an entire file that is your own code. Understood about jshint giving trouble over the capitalized SVGInjector but instead of ignoring a whole file, you can ignore one line by adding this commented out code next to the line of code `// jshint ignore:line`. See here for documentation - http://jshint.com/docs/  (do a find on "ignore"). Thanks

pcsandford commented:
@stulees thanks! I have made the changes you suggested and added the svg-inject in the bower.json file.

stulees commented:
@pcsandford You should just be able to specify a version for svg injector instead of pointing to the git repo. Try changing it to `"svg-injector": "1.1.3"` in the bower file to get the latest version. Thanks

pcsandford commented:
@stulees thanks! I have updated it. I was looking through the repo and also the bower page to see what the number was but couldn't find it. After your comment I saw it is listed by the dist folder. Sorry!

stulees commented:
@pcsandford :+1: all good, thanks

pcsandford commented:
@stulees cool, thanks for all the help. I will squash the commits before I merge. Have a great weekend!
